addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -399,24 +399,77 @@ def _import_fattura_pa(self, tree, invoice):
         for body_tree in tree.xpath('//FatturaElettronicaBody'):
             if not first_run or not invoice:
                 # make sure all the iterations create a new invoice record (except the first which could have already created one)
                 invoice = self.env['account.move']
             first_run = False
 
-            # Type must be present in the context to get the right behavior of the _default_journal method (account.move).
-            # journal_id must be present in the context to get the right behavior of the _default_account method (account.move.line).
-            elements = tree.xpath('//CessionarioCommittente//IdCodice')
-            company = elements and self.env['res.company'].search([('vat', 'ilike', elements[0].text)], limit=1)
-            if not company:
-                elements = tree.xpath('//CessionarioCommittente//CodiceFiscale')
-                company = elements and self.env['res.company'].search([('l10n_it_codice_fiscale', 'ilike', elements[0].text)], limit=1)
+            # First we will determine the company and direction (incoming / outgoing) of the move.
+            # There are 2 cases:
+            # - Import from tax agency (cron).
+            #   Both directions and all companies are possible.
+            #   Thus we look for company identifiers (VAT, codice fiscale) in both the buyer and the seller XML sections
+            #   to determine the company and direction of the move.
+            #   In this case no `default_move_type` is found in the context.
+            # - User clicks an "Upload" button on the invoices or bills view.
+            #   Only one direction should be considered (depending on the button).
+            #   The company must be one of the allowed ones for the current user.
+            #   "Upload" button on "Vendor Bills" view: incoming move; (allowed) company identifiers must be in the buyer section.
+            #   "Upload" button on "Invoices" view: outgoing move; (allowed) company identifiers must be in the seller section.
+            #   In this case a `default_move_type` is in the context.
+            #   It determines whether we look at an incoming or outgoing move.
+
+            default_move_type = self.env.context.get('default_move_type')
+            if default_move_type is None:
+                incoming_possibilities = [True, False]
+            elif default_move_type in invoice.get_purchase_types(include_receipts=True):
+                incoming_possibilities = [True]
+            elif default_move_type in invoice.get_sale_types(include_receipts=True):
+                incoming_possibilities = [False]
+            else:
+                _logger.warning("Cannot handle default_move_type '%s'.", default_move_type)
+                continue
+
+            allowed_company_ids = self.env.context.get('allowed_company_ids')
```

Comment:
Sven, can you verify what allowed_companies get summoned up in a cron?
We just need to fire it up and have a debug / print at the start.

Reply:
Good point. 

For "normal" users it "just works".

I only tested with the admin user. Is it used in day to day operations?
There the search does not seem to be restricted by the `allowed_company_ids`.

Better safe than sorry? Or should I just remove it?

Reply:
@svfu-odoo For that admin user, it might be normal (that it bypasses security rules) and there we need to check all companies anyways and for the user, the domain is added implicity.

Reply:
But we are searching on companies and users that have access to settings (like administrators) will be able to see all companies there even if they have not activated them in the settings.